subsys/bluetooth/controller/ll_sw/ull_adv.c

```diff
@@ -1723,7 +1723,7 @@ uint8_t ull_adv_data_set(struct ll_adv_set *adv, uint8_t len,
 	uint8_t idx;
 
 	/* Check invalid AD Data length */
-	if (len > PDU_AC_DATA_SIZE_MAX) {
+	if (len > PDU_AC_LEG_DATA_SIZE_MAX) {
```

Comment:
`ull_adv_data_set` seems to also be used when setting extended advertising data; checking just the legacy data size max seems odd to do in that case.

Reply:
After checking again, you are correct :)